core/src/main/java/org/infinispan/stream/StreamMarshalling.java

```diff
@@ -108,25 +132,24 @@ public V apply(Map.Entry<K, V> kvEntry) {
    }
 
    public static final class StreamMarshallingExternalizer implements AdvancedExternalizer<Object> {
-
-      private static final int EQUALITY_PREDICATE = 0;
-      private static final int ENTRY_KEY_FUNCTION = 1;
-      private static final int ENTRY_VALUE_FUNCTION = 2;
-      private static final int NON_NULL_PREDICATE = 3;
+      enum ExternalizerId {
+         EQUALITY_PREDICATE, ENTRY_KEY_FUNCTION, ENTRY_VALUE_FUNCTION, NON_NULL_PREDICATE, ALWAYS_TRUE_PREDICATE;
+      }
```

Comment:
@tristantarrant `ListExternalizer` also uses int ids... 

In the end, I really don't think `switch (ExternalizerId.values()[number])` is any better than `switch (number)`. And it also makes the `default` case useless, because an invalid number will cause an `ArrayIndexOutOfBoundsException`.